core/src/main/java/com/netflix/conductor/service/ExecutionService.java

```diff
@@ -342,14 +343,21 @@ private boolean requeue(Task pending) {
 
 	public List<Workflow> getWorkflowInstances(String workflowName, String correlationId, boolean includeClosed, boolean includeTasks) {
 
-		List<Workflow> workflows = executionDAOFacade.getWorkflowsByCorrelationId(correlationId, includeTasks);
+		List<Workflow> workflows = executionDAOFacade.getWorkflowsByCorrelationId(workflowName, correlationId, false);
 		List<Workflow> result = new LinkedList<>();
-		for (Workflow wf : workflows) {
-			if (wf.getWorkflowName().equals(workflowName) && (includeClosed || wf.getStatus().equals(Workflow.WorkflowStatus.RUNNING))) {
-				result.add(wf);
+		result = workflows.stream().parallel().filter(wf -> {
+			if (includeClosed || wf.getStatus().equals(Workflow.WorkflowStatus.RUNNING)) {
+				// including tasks for subset of workflows to increase performance
+				if (includeTasks) {
+					List<Task> tasks = executionDAOFacade.getTasksForWorkflow(wf.getWorkflowId());
+					tasks.sort(Comparator.comparingLong(Task::getScheduledTime).thenComparingInt(Task::getSeq));
```

Comment:
`Comparator.comparingLong(Task::getScheduledTime)` is not needed here since sorting by `seq` here would arrange the tasks in the correct order.